source/event_stream.c

```diff
@@ -749,111 +750,283 @@ struct aws_event_stream_header_value_pair aws_event_stream_create_int32_header(
+        .value_owned = 0,
+        .header_value_type = AWS_EVENT_STREAM_HEADER_TIMESTAMP,
+    };
+
+    aws_write_u64((uint64_t)value, header.header_value.static_val);
+
+    memcpy((void *)header.header_name, (void *)name.ptr, (size_t)name.len);
+
+    return aws_array_list_push_back(headers, (void *)&header);
+}
+
+int aws_event_stream_add_uuid_header_by_cursor(
+    struct aws_array_list *headers,
+    struct aws_byte_cursor name,
+    struct aws_byte_cursor value) {
+
+    AWS_FATAL_PRECONDITION(headers);
+    AWS_FATAL_PRECONDITION(name.len > 0);
+    AWS_FATAL_PRECONDITION(name.ptr != NULL);
+    AWS_RETURN_ERROR_IF(
+        name.len <= AWS_EVENT_STREAM_HEADER_NAME_LEN_MAX, AWS_ERROR_EVENT_STREAM_MESSAGE_INVALID_HEADERS_LEN);
+    AWS_RETURN_ERROR_IF(value.len == UUID_LEN, AWS_ERROR_EVENT_STREAM_MESSAGE_INVALID_HEADERS_LEN);
+
+    struct aws_event_stream_header_value_pair header = {
+        .header_name_len = (uint8_t)name.len,
+        .header_value_len = UUID_LEN,
+        .value_owned = 0,
+        .header_value_type = AWS_EVENT_STREAM_HEADER_UUID,
+    };
+
+    memcpy((void *)header.header_name, (void *)name.ptr, (size_t)name.len);
+    memcpy((void *)header.header_value.static_val, value.ptr, UUID_LEN);
+
+    return aws_array_list_push_back(headers, (void *)&header);
+}
+
+int aws_event_stream_add_byte_header(struct aws_array_list *headers, const char *name, uint8_t name_len, int8_t value) {
+    struct aws_byte_cursor name_cursor = {
+        .ptr = (uint8_t *)name,
+        .len = (size_t)name_len,
+    };
+
+    return aws_event_stream_add_byte_header_by_cursor(headers, name_cursor, value);
+}
+
+int aws_event_stream_add_bool_header(struct aws_array_list *headers, const char *name, uint8_t name_len, int8_t value) {
+    struct aws_byte_cursor name_cursor = {
+        .ptr = (uint8_t *)name,
+        .len = (size_t)name_len,
+    };
+
+    return aws_event_stream_add_bool_header_by_cursor(headers, name_cursor, value != 0);
+}
+
+int aws_event_stream_add_int16_header(
+    struct aws_array_list *headers,
+    const char *name,
+    uint8_t name_len,
+    int16_t value) {
+
+    struct aws_byte_cursor name_cursor = {
+        .ptr = (uint8_t *)name,
+        .len = (size_t)name_len,
+    };
+
+    return aws_event_stream_add_int16_header_by_cursor(headers, name_cursor, value);
+}
+
+int aws_event_stream_add_int32_header(
+    struct aws_array_list *headers,
+    const char *name,
+    uint8_t name_len,
+    int32_t value) {
+
+    struct aws_byte_cursor name_cursor = {
+        .ptr = (uint8_t *)name,
+        .len = (size_t)name_len,
+    };
+
+    return aws_event_stream_add_int32_header_by_cursor(headers, name_cursor, value);
+}
+
+int aws_event_stream_add_int64_header(
+    struct aws_array_list *headers,
+    const char *name,
+    uint8_t name_len,
+    int64_t value) {
+
+    struct aws_byte_cursor name_cursor = {
+        .ptr = (uint8_t *)name,
+        .len = (size_t)name_len,
+    };
```

Comment:
trivial: just FYI, these could also be `aws_byte_cursor_from_array(name, (size_t)name_len)`